projects/aca-playwright-shared/src/page-objects/components/actions-dropdown.component.ts

```diff
@@ -52,10 +52,10 @@ export enum ActionType {
 export class ActionsDropdownComponent extends BaseComponent {
   private static rootElement = 'aca-rule-action-list';
 
-  private getOptionLocator = (optionName: string): Locator => this.page.locator('.mat-select-panel .mat-option-text', { hasText: optionName });
+  private getOptionLocator = (optionName: string): Locator => this.page.locator('.mat-select-panel .mat-option-text', { hasText: optionName }).first();
   private ruleActionLocator = this.getChild('aca-rule-action [data-automation-id="rule-action-card-view"]');
   private addActionButtonLocator = this.getChild('[data-automation-id="rule-action-list-add-action-button"]');
-  private actionDropdownLocator = this.getChild('[data-automation-id="rule-action-select"]');
+  private actionDropdownLocator = this.page.locator('//mat-select//span[text()="Select an action"]');
```

Comment:
this is unsafe, as relies on some path that can change any time, use exact elements instead

Reply:
new uesed data-automation-id more stable